src/lcm/discrete_problem.py

```diff
@@ -0,0 +1,280 @@
+"""Functions that aggregate the conditional continuation values over discrete choices.
+
+By conditional continuation value we mean continuation values conditional on a discrete
+choice, i.e. the result of solving the continuous choice problem conditional on the
+discrete choice. These are also _conditional_ on a given state.
+
+By aggregate we mean calculating the expected maximum of the continuation values, based
+on the distribution of utility shocks. Currently we support no shocks. In the future,
+we will at least support IID Extreme Value Type 1 shocks.
+
+How we (want to) solve the problem:
+-----------------------------------
+
+- No shocks: We take the maximum of the conditional continuation values.
+
+- IID Extreme Value Type 1 shocks: We do a logsum calculation.
+
+"""
+
+from collections.abc import Callable
+from functools import partial
+from typing import Any, Literal
+
+import jax
+import jax.numpy as jnp
+import pandas as pd
+from jax import Array
+
+from lcm.typing import SegmentInfo
+
+
+def get_solve_discrete_problem(
+    *,
+    shock_type: Literal["extreme_value"] | None,
```

Comment:
```suggestion
    random_utility_shock_type: Literal["extreme_value"] | None,
```
Better be explicit here, there might be other types of randomness within period, as just discussed.